src/components/sidebar/MarkerSidebar.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+
+import explicitConnect from '../../utils/connect';
+import { selectedThreadSelectors } from '../../reducers/profile-view';
+import { getSelectedThreadIndex } from '../../reducers/url-state';
+import { formatMilliseconds } from '../../utils/format-numbers';
+import { CanSelectContent } from './index';
+
+import type {
+  ConnectedProps,
+  ExplicitConnectOptions,
+} from '../../utils/connect';
+import type { ThreadIndex } from '../../types/profile';
+import type {
+  IndexIntoTracingMarkers,
+  TracingMarker,
+} from '../../types/profile-derived';
+
+type SidebarDetailProps = {|
+  +label: string,
+  +value: React.Node,
+|};
+
+function SidebarDetail({ label, value }: SidebarDetailProps) {
+  return (
+    <React.Fragment>
+      <div className="sidebar-label">{label}:</div>
+      <div className="sidebar-value">{value}</div>
+    </React.Fragment>
+  );
+}
+
+type StateProps = {|
+  +selectedNodeIndex: IndexIntoTracingMarkers,
+  +selectedThreadIndex: ThreadIndex,
+  +markers: TracingMarker[],
+|};
+
+type Props = ConnectedProps<{||}, StateProps, {||}>;
+
+class MarkerSidebar extends React.PureComponent<Props> {
+  render() {
+    const { selectedNodeIndex, markers } = this.props;
+
+    if (selectedNodeIndex === null || selectedNodeIndex === -1) {
```

Comment:
as a quick note of history, this `-1` value comes from a long time in the project history :) in the future we'll want to use `null`.